### ISSUE TITLE
[Feature]: Reorganise the error log meta data

### ISSUE THREAD

**Issue by sribalajig** [appsmithorg/appsmith#17239]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Here is a snapshot of an error that I faced while executing a query -

![better_formatting_errors](https://user-images.githubusercontent.com/15727421/193273965-30c5b670-7cdb-4609-9d4b-29c8a999a738.png)

Here are some issues with the logged values -

1. `actionId` is not something I am interested in knowing.
2. The `requestedAt` property is confusing - is it a timestamp? Is so, this is also not really relevant to put here because I already have the timestamp on the top left.
3. The query and parameters passed are things I am actually interested in knowing, but these are embedded in a JSON dump and I need to expand them to get more details.

As an alternative to logging a JSON object which is difficult to read, can we have a better representation of the error which gives users the following details -

1. What is the error in plain English?
2. What is the stack trace?
3. What is the detailed error coming from my datasource?
4. What can I do to fix it?

### Why should this be worked on?

If we have cleanly formatted error logs with relevant/contextual information, along with directions on how to fix the error, we can have users self serve and fix errors themselves.

**Comment by torcoste** [comment]
Great points, @sribalajig! 👏

@Nikhil-Nandagopal, is the team interested in improving error handling?
